ord_schema/validations.py

```diff
@@ -151,7 +171,11 @@ def validate_message(message, recurse=True, raise_on_error=True):
             f"Don't know how to validate {type(message)}")
 
     with warnings.catch_warnings(record=True) as tape:
-        _VALIDATOR_SWITCH[type(message)](message)
+        if validate_ids and isinstance(
+                message, (reaction_pb2.Reaction, dataset_pb2.Dataset)):
+            _VALIDATOR_SWITCH[type(message)](message, validate_id=True)
```

Comment:
I don't _love_ how this turned out, but it was the most elegant way I could think of to have an easy switch between...
1. Validating where we don't care if IDs are malformed
1. Validating where we want to raise an error if IDs are malformed

Reply:
LGTM, although I have a slight preference for the default being (2) and needing to explicitly disable it for (1). This matches the default strictness of `raise_on_error=True`.

Reply:
I would advocate for a default of False, since a normal use case will be people checking their programmatic definitions for validity before submission. Having the default be True might encourage users to assign their own IDs so the validation passes, which I don't think we want to do